### PR TITLE
Provide recipient option `default` in flow mail send modal if trigger implements `MailAware`

### PR DESCRIPTION
**Pull request by tgauges**
### 1. Why is this change necessary?

See [3260](https://github.com/shopware/platform/issues/3260)

### 2. What does this change do, exactly?

This change makes it possible to configure an 'Automatic' email recipient in a send mail flow action if the flow trigger implements `MailAware`.

### 3. Describe each step to reproduce the issue or behaviour.

See [3260](https://github.com/shopware/platform/issues/3260)

### 4. Please link to the relevant issues (if any).

See [3260](https://github.com/shopware/platform/issues/3260)

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [ ] I have written tests and verified that they fail without my change
- [ ] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfil them.

### <samp>🤖 Generated by Copilot at 369c675</samp>

Added a new feature to the flow module that allows users to choose the mail structure for mail aware entities in the mail send modal. Added the corresponding snippets for the German and English translations. Refactored some code for readability.

### <samp>🤖 Generated by Copilot at 369c675</samp>

*  Add `recipientMailStruct` computed property to return mail structure option for mail send modal ([link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-44642660d6e8c75eea1dead363d11c0b7011a733be9e65bc01b806d9a5e8c07bR68-R76))
*  Add `mailAware` computed property to return mail aware entities for mail send modal ([link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-44642660d6e8c75eea1dead363d11c0b7011a733be9e65bc01b806d9a5e8c07bR117-R120))
*  Modify `recipientOptions` computed property to conditionally include mail structure option based on mail aware entities ([link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-44642660d6e8c75eea1dead363d11c0b7011a733be9e65bc01b806d9a5e8c07bR163-R172))
*  Add snippet for mail structure option label to German and English translation files ([link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-9c6aa1c7adeb73b7549f8b772686874e5e9135378e376a6c12d284c5bb189689R244), [link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-cd85183509c27c812abafd565ba3593d4d90b9c0ae2fd791d3d6644576a6beddR244))
*  Reformat code to remove spaces inside curly braces for consistency and readability ([link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-44642660d6e8c75eea1dead363d11c0b7011a733be9e65bc01b806d9a5e8c07bL1-R7), [link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-44642660d6e8c75eea1dead363d11c0b7011a733be9e65bc01b806d9a5e8c07bL407-R431), [link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-44642660d6e8c75eea1dead363d11c0b7011a733be9e65bc01b806d9a5e8c07bL415-R438), [link](https://github.com/shopware/platform/pull/3261/files?diff=unified&w=0#diff-44642660d6e8c75eea1dead363d11c0b7011a733be9e65bc01b806d9a5e8c07bL502-R525))

### CODE REVIEW

**Comment by tgauges** [comment]
There might be changes necessary to <https://docs.shopware.com/en/shopware-6-en/settings/Flow-Builder#action>?

**Comment by tgauges** [comment]
Before I start writing tests, does this feature have a chance of being merged?

**Comment by vienthuong** [comment]
Hi @tgauges, 
First of all, thanks for your contribution.
The changes look good to me, could you write the unit test for it?

File: src/Administration/Resources/app/administration/src/module/sw-flow/component/modals/sw-flow-mail-send-modal/index.js [hunk 1]
```diff
@@ -147,6 +160,16 @@ export default {
                 ];
             }
 
+            const hasMailAware = allowAwareConverted.some(allowedAware => this.mailAware.includes(allowedAware));
+
+            if (hasMailAware) {
```
Comment (larskemper) [inline src/Administration/Resources/app/administration/src/module/sw-flow/component/modals/sw-flow-mail-send-modal/index.js:165]: Please add a test case

**Review by larskemper — CHANGES_REQUESTED** [review]

**Comment by larskemper** [comment]
Hey @tgauges :wave:, nice work :muscle:.
Please sign the [CLA](https://cla-assistant.io/shopware/shopware?pullRequest=3261), resolve the open threads and create a changelog file so we can merge your changes

**Comment by tgauges** [comment]
Sadly I don't work with with the project anymore which needs this patch. So I currently don't have the time to write the tests.

But I signed the CLA.

**Comment by larskemper** [comment]
I will close this PR due to inactivity, feel free to reopen it whenever you worked on the tests ✌️

**Pull request closed without merging**